sips/committee_consensus.md

```diff
@@ -0,0 +1,486 @@
+
+### Omitting Partial Signatures
+
+If in post-consensus stage for attestation duty, the duty's slot is lower than a validator's `highestAttestingSlot`, the `CommitteeRunner` will omit the post-consesnus message for this validator. 
+
+### Cutoff Round
+We will also create a better Cutoff Round.
+
+#### Sync Committee
+`CutOffRound = 4  \\ one slot`
+
+#### Attestation
+`CutOffRound = 12 \\ one epoch`
+
+### CommitteeID
+
+An identifier for committee must be added to `MessageID`.
+
+```go
+type CommitteeID [32]byte
+
+// Return a 32 bytes ID for the committee of operators
+func getCommitteeID(committee []OperatorID) CommitteeID {
+	// sort
+	sort.Slice(committee, func(i, j int) bool {
+		return committee[i] < committee[j]
+	})
+	// Convert to bytes
+	bytes := make([]byte, len(committee)*4)
+	for i, v := range committee {
+		binary.LittleEndian.PutUint32(bytes[i*4:], uint32(v))
+	}
+	// Hash
+	return CommitteeID(sha256.Sum256(bytes))
+}
+```
+
+In order to route consensus messages to the correct consensus runner, the `CommitteeID` field will be included in the `MessageID` replacing `ValidatorPublicKey`.
+
+### Role
+
+This is used to route the message to the correct runner. Since `CommitteeRunner` has several beacone roles we will create a new `RunnerRole` in the MessageID.
+
+```go
+// TODO since this is on wire no real need to take 32 bits
+type RunnerRole int32
+
+const (
+	RoleCommittee RunnerRole = iota
+	RoleAggregator
+	RoleProposer
+	RoleSyncCommitteeContribution
+
+	RoleValidatorRegistration
+	RoleVoluntaryExit
+
+	RoleUnknown = -1
+)
+```
+
+### MessageID
+
+`ValidatorPublicKey` and `CommitteeID` will be used interchangeably in `MessageID`. `Role` will change location because it is used to determine ID type.  `CommitteeID` is 32 bytes long so it will be encoded with a `0x00` 16 bytes prefix to make it the same length as `ValidatorPublicKey`.
+
+```go
+const (
+	domainSize     = 4
+	domainStartPos = 0
+	roleTypeSize   = 4
+	// CHANGE IN Positions
+	roleTypeStartPos = domainStartPos + domainSize
+	senderIDSize     = 48
+	senderIDStartPos = roleTypeStartPos + roleTypeSize
+)
+
+// MessageID is used to identify and route messages to the right validator and Runner
+type MessageID [56]byte
+
+func (msg MessageID) GetDomain() DomainType {
+	return msg[domainStartPos : domainStartPos+domainSize]
+}
+
+func (msg MessageID) GetRoleType() RunnerRole {
+	roleByts := msg[roleTypeStartPos : roleTypeStartPos+roleTypeSize]
+	return BeaconRole(binary.LittleEndian.Uint32(roleByts))
+}
+
+func (msg MessageID) GetSenderID() []byte {
+	return msg[senderIDStartPos : senderIDStartPos+senderIDSize]
+}
+
+func parseMessageID(messageID MessageID) (DomainType, BeaconRole, []byte) {
+	domain := messageID.GetDomain()
+	role := messageID.GetRoleType()
+	if role == BNRoleSyncCommitteeContribution {
+		return domain, BNRoleAttesterOrSyncCommittee, messageID.GetSenderID()[16:]
```

Comment:
if there's no special reason for putting committee ID at the end of sender ID, i'd recommend the other way around due to it being less of a "surprise" to the common pattern of taking up the left side of bigger than needed arrays.

plus, it'd make such conversions correct by default (no magic number is required):

```go
cid := CommitteeID(senderID)
```

Reply:
This won't work due to 0x00 prefix